num_to_add[] triple_vect
---
int32[] sum_vect
